mobile/lib/utils/extensions.dart

```diff
@@ -143,25 +143,25 @@ extension KyaListExt on List<Kya> {
     });
   }
 
-  List<Kya> filterInProgressKya() {
+  List<KyaLesson> filterInProgressKya() {
     return where((element) {
       return element.isInProgress();
     }).toList();
   }
 
-  List<Kya> filterToDo() {
+  List<KyaLesson> filterToDo() {
     return where((element) {
       return element.todo();
     }).toList();
   }
 
-  List<Kya> filterPendingCompletion() {
+  List<KyaLesson> filterPendingCompletion() {
     return where((element) {
       return element.isPendingCompletion();
     }).toList();
   }
 
-  List<Kya> filterComplete() {
+  List<KyaLesson> filterComplete() {
     return where((element) {
       return element.isComplete();
     }).toList();
```

Comment:
Is there a way we can handle this functionality in the backend and assign statuses to the returned Lessons rather than computing the status within the app?? @BenjaminSsempala